johnsonandjohnson/Bodiless-JS
# feat(layouts): Component Switcher

## Changes

- Adds the ability to switch out a component in the flexbox while retaining the same data.
- Refactor of flexbox to use more standard bodiless hoc integrations
- Causes insert item action to insert new item after the current item 
- Causes delete item action to focus the adjacent item rather than the flexbox itself (unless flexbox is empty).

## Test Instructions

- On homepage of test site click on one of the touts and then click the "switch" button -- replace with another, similar, tout.  Verify that the content (for whatever fields exist on both touts) remains the same.
- Delete a tout.  Verify that the other tout has focus
- Delete the other tout, verify that the flexbox has focus (+ button avilable).
- Add back some touts
- insert a tout when first tout is focused - verify that new tout is added in the middle.

## Related Issues

Closes #69
